gh-140166: Use `application/texinfo` as sole MIME type for `.texi` and `.texinfo` files (#140165)
python/cpython: Misc/NEWS.d/next/Library/2025-10-15-15-10-34.gh-issue-140166.NtxRez.rst

> Per https://www.iana.org/assignments/media-types/application/texinfo, the .texi and .texinfo file formats has an officially-registered MIME type, thus deprecating the application/x-texinfo alias.

Reviewer: Rather than saying deprecating, which suggests keeping the old one had having deprecation warnings/notices, let's rather say we're replacing the old one with the new one.

> Per https://www.iana.org/assignments/media-types/application/texinfo, the .texi and .texinfo file formats has an officially-registered MIME type, application/texinfo Therefore remove the outdated application/x-texinfo alias.

Reviewer 2: ```suggestion
:mod:`mimetypes`: Per the `IANA assignment <https://www.iana.org/assignments/media-types/application/texinfo>`_, update the MIME type for the ``.texi`` and ``.texinfo`` file formats to ``application/texinfo``, instead of ``application/x-texinfo``.
```
Little rewording suggestion.
